internal/provisioner/kops_provisioner.go

```diff
@@ -233,12 +255,85 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster) error
 		return err
 	}
 
+	// Old tiller pod is deleted because of the upgrade action and therefore wait is needed
+	logger.Infof("Waiting %d seconds for the old Tiller pod to be replaced...", wait)
+	time.Sleep(time.Duration(wait) * time.Second)
+
+	// Begin deploying Helm charts
+	privateNginx := helmDeployment{valuesPath: "helm-charts/private-nginx_values.yaml", chartName: "stable/nginx-ingress", namespace: "internal-nginx", chartDeploymentName: "private-nginx"}
+	prometheus := helmDeployment{valuesPath: "helm-charts/prometheus_values.yaml", chartName: "stable/prometheus", namespace: "prometheus", chartDeploymentName: "prometheus-client", dns: (cluster.ID + ".prometheus." + provisioner.privateDNS)}
+
+	helmDeployments := []helmDeployment{privateNginx, prometheus}
+
+	for _, value := range helmDeployments {
+		err = helmInstallation(value, logger, kops)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Get the new ELB internal-nginx endpoint
+	endpoint, err := getEndpoint("internal-nginx", logger, kops)
+	if err != nil {
+		return err
+	}
+
+	for _, app := range helmApps {
+		logger.Infof("Registering DNS for %s", app)
+		dns := fmt.Sprintf("%s.%s.%s", cluster.ID, app, provisioner.privateDNS)
+		err = provisioner.aws.CreateCNAME(dns, []string{endpoint}, logger)
+		if err != nil {
+			return err
+		}
+	}
+
 	logger.WithField("name", kopsMetadata.Name).Info("Successfully created cluster")
 
 	return nil
 }
 
-// HelmSetup is used for the initial setup of Helm in cluster
+// getEndpoint is used to get the endpoint of the internal ingress.
+func getEndpoint(namespace string, logger log.FieldLogger, kops *kops.Cmd) (string, error) {
+	k8sClient, err := k8s.New(kops.GetKubeConfigPath(), logger)
+	if err != nil {
+		return "", err
+	}
+	services, err := k8sClient.GetServices(namespace)
+	if err != nil {
+		return "", err
+	}
+	for _, service := range services.Items {
+		if service.Status.LoadBalancer.Ingress != nil {
+			endpoint := service.Status.LoadBalancer.Ingress[0].Hostname
+			logger.Infof("Succesfully got LoadBalancer endpoint %s for Namespace %s", endpoint, namespace)
+			return endpoint, nil
+		}
+	}
+	return "", nil
+}
+
+// helmInstallation is used to install Helm charts.
+func helmInstallation(chart helmDeployment, logger log.FieldLogger, kops *kops.Cmd) error {
```

Comment:
This could probably have a more descriptive. Maybe `installHelmChart()`?

Reply:
Also, we are passing in a pointer to kops.Cmd here, but all we need it for is the config path, so we should just pass in the path string instead.

```suggestion
func helmInstallation(chart helmDeployment, logger log.FieldLogger, configPath string) error {
```

This should be done for the other functions for helm if they only need the config path string as well.

Reply:
Fixed name plus to pass only the configpath needed.